src/v/config/configuration.cc

```diff
@@ -173,7 +173,19 @@ configuration::configuration()
 
   , min_version(*this, "min_version")
   , max_version(*this, "max_version")
-
+  , max_recovery_memory(
```

Comment:
Would be good to prefix the config name with a subsystem and pull nouns to front to get neat alphabetical sorting: like `raft_recovery_max_memory` so that it'll go next to `raft_recovery_default_read_size`